Cue/app/reducers/library.js

```diff
@@ -162,46 +162,129 @@ function library(state: State = initialState, action: Action): State {
 //apply card changes to a deck.cards
 function _applyCardChangesDecks(cards, changes) {
   console.debug('_applyCardChangesDecks', cards, changes)
-  if (! cards && !changes) return []
+  if (!cards && !changes) return []
   if (!changes) return cards
   if (!cards) cards = []
+
+  let positionUpdateNeeded = false
   let mergedCards = cards.slice();
+
   changes.forEach(change => {
     let index = mergedCards.findIndex(card => card.uuid == change.uuid)
+
     if (change.action === 'add') {
       mergedCards.push({...change, action: undefined})
-    } else if (change.action === 'delete' && mergedCards[index]) {
-      mergedCards.splice(index,1)
-    } else if (change.action === 'edit' && mergedCards[index]) {
-      //TODO: update the position of every card if position changed
-      mergedCards[index] = {...mergedCards[index], ...change, action: undefined}
+
+    } else if (change.action === 'delete') {
+      if (index !== -1) {
+        mergedCards.splice(index,1)
+      } else {
+        console.warn('Tried to delete non-existant card ' + change.uuid + '?')
+      }
+
+    } else if (change.action === 'edit') {
+      if (index !== -1) {
+        if (typeof change.position !== 'undefined' && change.position !== index) {
+          // The change has a move; do the move
+          positionUpdateNeeded = true
+          let card = mergedCards.splice(index, 1)[0]
+          card = {...card, ...change, action: undefined}
```

Comment:
Yeah we want to remove the action from the change so that it doesn't appear in the resulting merged card.